packages/@ember/-internals/runtime/lib/mixins/-proxy.js

```diff
@@ -85,6 +87,10 @@ export default Mixin.create({
     }
   },
 
```

Comment:
note that things like `ember-m3` use `unknownProperty` but do not extend `Ember.ObjectProxy`, we'll need to make sure we have some mechanism for that situation.

Reply:
Yeah, this isn't meant to be public just yet, but I think that this is getting closer to the shape of solution for M3. I think we still have all of Ember's _standard_ model use-cases covered too.